### ISSUE TITLE
[BUG] Regression bug causing ccextractor 0.94 to crash

### ISSUE THREAD

**Issue by rboy1** [CCExtractor/ccextractor#1575]
CCExtractor version: 0.94

# In raising this issue, I confirm the following:

- [X ] I have read and understood the [contributors guide](https://github.com/CCExtractor/ccextractor/blob/master/.github/CONTRIBUTING.md).
- [ X] I have checked that the bug-fix I am reporting can be replicated, or that the feature I am suggesting isn't already present.
- [ X] I have checked that the issue I'm posting isn't already reported.
- [ X] I have checked that the issue I'm porting isn't already solved and no duplicates exist in [closed issues](https://github.com/CCExtractor/ccextractor/issues?q=is%3Aissue+is%3Aclosed) and in [opened issues](https://github.com/CCExtractor/ccextractor/issues)
- [ X] I have checked the pull requests tab for existing solutions/implementations to my issue/suggestion.
- [ X] I have used the latest available version of CCExtractor to verify this issue exists.
- [ X] I have ticked all the boxes in this section and to prove it I'm deleting the section completely to remove boilerplate text.

# Necessary information

- Is this a regression (i.e. did it work before)? YES, it works fine with 0.93
- What platform did you use? Windows 10 x64
- What were the used arguments? `EIA-608 and EIA-708 streams.ts`

# Video links
https://www.dropbox.com/scl/fi/zoaavo5y9szk1tn4uee7l/EIA-608-and-EIA-708-streams.ts?rlkey=46xgwv6rdamo50mpnxk7x9eux&dl=0

# Additional information

Using `set RUST_BACKTRACE=full` to get a full stack trace, see below.

Here's the output of the CLI

```
ccextractorwinfull.exe "EIA-608 and EIA-708 streams.ts"

CCExtractor 0.94, Carlos Fernandez Sanz, Volker Quetschke.
Teletext portions taken from Petr Kutalek's telxcc
--------------------------------------------------------------------------
Input: EIA-608 and EIA-708 streams.ts
[Extract: 1] [Stream mode: Autodetect]
[Program : Auto ] [Hauppage mode: No] [Use MythTV code: Auto]
[CEA-708: 63 decoders active]
[CEA-708: using charset "none" for all services]
[Timing mode: Auto] [Debug: No] [Buffer input: Yes]
[Use pic_order_cnt_lsb for H.264: No] [Print CC decoder traces: No]
[Target format: .srt] [Encoding: UTF-8] [Delay: 0] [Trim lines: No]
[Add font color data: Yes] [Add font typesetting: Yes]
[Convert case: No][Filter profanity: No] [Video-edit join: No]
[Extraction start time: not set (from start)]
[Extraction end time: not set (to end)]
[Live stream: No] [Clock frequency: 90000]
[Teletext page: Autodetect]
[Start credits text: None]
[Quantisation-mode: CCExtractor's internal function]

-----------------------------------------------------------------
Opening file: EIA-608 and EIA-708 streams.ts
File seems to be a transport stream, enabling TS mode
Analyzing data in general mode
  1%  |  00:00

New video information found
[1920 * 1080] [AR: 03 - 16:9] [FR: 04 - 29.97] [progressive: no]

XDS Program Type: [21-Entertainment]
thread '<unnamed>' panicked at 'index out of bounds: the len is 1 but the index is 1', src\decoder\service_decoder.rs:275:25
stack backtrace:
   0:   0x4c862a - <unknown>
   1:   0x535581 - <unknown>
   2:   0x538ca8 - <unknown>
   3:   0x538820 - <unknown>
   4:   0x53928e - <unknown>
   5:   0x538e86 - <unknown>
   6:   0x537aef - <unknown>
   7:   0x4cf102 - <unknown>
   8:   0x4c1381 - <unknown>
   9:   0x6cb295 - <unknown>
  10:   0x6f6aba - <unknown>
  11:   0x6e02d4 - <unknown>
  12:   0x6df3d3 - <unknown>
  13:   0x6dffa1 - <unknown>
  14:   0x6e30be - <unknown>
  15:   0x6e2954 - <unknown>
  16:   0x65f53f - <unknown>
  17:   0x65ff55 - <unknown>
  18:   0x995687 - <unknown>
  19: 0x763afcc9 - BaseThreadInitThunk
  20: 0x77187c6e - RtlGetAppContainerNamedObjectPath
  21: 0x77187c3e - RtlGetAppContainerNamedObjectPath
```

It works fine with 0.93

**Comment by IshanGrover2004** [comment]
I guess this issue is fixed in [code](https://github.com/CCExtractor/ccextractor/blob/376ff831616919e092b53353f3799654d86d0759/src/rust/src/decoder/service_decoder.rs#L274) after 0.94 package release.
So, upcoming release doesn't contain this bug

**Comment by rboy1** [comment]
> I guess this issue is fixed in [code](https://github.com/CCExtractor/ccextractor/blob/376ff831616919e092b53353f3799654d86d0759/src/rust/src/decoder/service_decoder.rs#L274) after 0.94 package release. So, upcoming release doesn't contain this bug

The last update to this code was 2 years ago. Are you sure it's fixed?

**Comment by IshanGrover2004** [comment]
> The last update to this code was 2 years ago. Are you sure it's fixed?

Yeah, I also checked that but I ran `./ccextracter sample.ts -608` from source binary which run successfully whereas with installed binary its not running(panicking because of `index out of bounds`).

**Comment by rboy1** [comment]
Is that the same sample I uploaded: https://www.dropbox.com/scl/fi/zoaavo5y9szk1tn4uee7l/EIA-608-and-EIA-708-streams.ts?rlkey=46xgwv6rdamo50mpnxk7x9eux&dl=0

**Comment by IshanGrover2004** [comment]
> Is that the same sample I uploaded: https://www.dropbox.com/scl/fi/zoaavo5y9szk1tn4uee7l/EIA-608-and-EIA-708-streams.ts?rlkey=46xgwv6rdamo50mpnxk7x9eux&dl=0

Yes, i used the same sample you uploaded
And this is the [output.srt](https://pastebin.com/wAiHYmKW)

**Comment by IshanGrover2004** [comment]
Do you know, what could be problem here?
@cfsmp3
